api/web/base.py

```diff
@@ -310,6 +341,39 @@ def handle_exception(self, exception, debug):
 
         util.send_json_http_exception(self.response, str(exception), code, custom=custom_errors)
 
+    def log_user_access(self, access_type, cont_name=None, cont_id=None):
+
+        log_map = {
+            'access_type':      access_type.value,
+            'request_method':   self.request.method,
+            'request_path':     self.request.path,
+            'origin':           self.origin,
+            'timestamp':        datetime.datetime.utcnow()
+        }
+
+        if access_type not in [AccessType.user_login, AccessType.user_logout]:
```

Comment:
Before this line, do a type check on `access_type` to confirm that it's truly an enum (else throw).